ca/django_ca/utils.py

```diff
@@ -845,7 +845,7 @@ def parse_general_name(name: ParsableGeneralName) -> x509.GeneralName:
         if match is not None:
             oid, asn_typ, val = match.groups()
             if asn_typ == "UTF8":
-                parsed_value = val.encode("utf-8")
+                parsed_value = UTF8String(val)
```

Comment:
Closes https://github.com/mathiasertl/django-ca/issues/76

Reply:
Cool - can you please add some test-cases that test this? This really is a part of the standard that I barely use - hence the issues of course.  `ca/django_ca/tests/tests_utils.py` already has many test cases, you can just add some there.

Reply:
I modified the existing test – I'm not too familiar with `unittest` though. Especially since `django-ca`s test suite is impressively extensive – e.g. automatic linting and different deployment test scenarios are still on my "to look into & learn"-list. I think implementing an integration test to see the Certificate is actually signed could be desirable here too.

Reply:
> I think implementing an integration test to see the Certificate is actually signed could be desirable here too.

Yes! Check out [this test|https://github.com/mathiasertl/django-ca/blob/master/ca/django_ca/tests/tests_command_sign_cert.py#L327] here. `cmd_e2e` lets you pass the command line as you would give it on the command line. It will pass through the entire argparse stack and everything (unlike the standard Django `cmd()` function).  The test already passes an alt name - you could implement a similar test where you pass an "advanced" SAN and then check if the resulting cert contains the expected name.